ee/server/service/software_installers.go

```diff
@@ -236,3 +236,16 @@ func (svc *Service) InstallSoftwareTitle(ctx context.Context, hostID uint, softw
 
 	return nil
 }
+
+func (svc *Service) GetSoftwareInstallResults(ctx context.Context, resultUUID string) (*fleet.HostSoftwareInstallerResult, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.HostSoftwareInstallerResultAuthz{}, fleet.ActionRead); err != nil {
```

Comment:
here you need to pass the `team_id` of the host associated with the given `resultUUID`, so you will need to do a bit of gymnastics here, check out the service methods for scripts or MDM commands for inspiration, but in general lines:

1. See if there's a general authorization check that can be done before calling the db method
2. Call `ds.GetSoftwareInstallResults` and either make it return the team id of the host, or, using the host ID, call `ds.HostLite` to get the team ID
3. Be careful about leaking information to non-authorized users when something is not found
4. Do the authz check using the host team id

Reply:
In case it is helpful, here's the [gymnastics](https://github.com/fleetdm/fleet/pull/18693/files#diff-b3883848dac3454f4ed4968a94f9ca335241f9f99bfad92f95a6bfb7157523ebR115) I did for downloading software installers.

Reply:
@jahzielv I think this check will still fail for people that are members of a team. Looking at the rego policy, if `HostSoftwareInstallerResultAuthz` doesn't have a `HostTeamID`, then you _must_ have a global role.

You probably want to check that the user can list hosts here.

Reply:
@roperzh done!